src/rust/engine/src/scheduler.rs

```diff
@@ -140,6 +142,33 @@ impl Scheduler {
     m
   }
 
+  ///
+  /// Returns references to all Python objects held alive by the graph, and a summary of sizes of
+  /// Rust structs as a count and total size.
+  ///
+  pub fn live_items(&self, session: &Session) -> (Vec<Value>, HashMap<String, (usize, usize)>) {
+    let context = Context::new(self.core.clone(), session.clone());
+    let mut items = vec![];
+    let mut sizes: HashMap<String, (usize, usize)> = HashMap::new();
+    // TODO: Creation of a Context is exposed in https://github.com/Aeledfyr/deepsize/pull/31.
+    let mut deep_context = deepsize::Context::new();
+    self.core.graph.visit_live(&context, |k, v| {
+      if let NodeKey::Task(ref t) = k {
+        items.extend(t.params.keys().map(|k| k.to_value()));
+        items.push(v.clone().try_into().unwrap());
+      }
+      let mut entry = sizes.entry(k.workunit_name()).or_insert_with(|| (0, 0));
+      entry.0 += 1;
+      entry.1 += {
+        std::mem::size_of_val(&k)
+          + k.deep_size_of_children(&mut deep_context)
+          + std::mem::size_of_val(&v)
+          + v.deep_size_of_children(&mut deep_context)
```

Comment:
Thoughts if we should be worried about overflow? Maybe use `checked_add`, and if it fails then log at ERROR level and ask people to write a bug report?

Reply:
Shouldn't need to worry about it: overflowing `usize` would require 18 exabytes of memory.